administrator/components/com_joomlaupdate/extract.php

```diff
@@ -0,0 +1,1670 @@
+	 * issue in the context of updating Joomla as the ZIP archive contains fairly small files.
+	 *
+	 * If this is too low it will cause too many requests to hit the server, potentially triggering a DoS protection and
+	 * causing the extraction to fail. If this is too big the extraction will not be as verbose and the user might think
+	 * something is broken. A value between 3 and 7 seconds is, therefore, recommended.
+	 *
+	 * @var int
+	 */
+	private const MAX_EXEC_TIME = 4;
+
+	/**
+	 * Run-time execution bias (percentage points).
+	 *
+	 * We evaluate the time remaining on the timer before processing each file on the ZIP archive. If we have already
+	 * consumed at least this much percentage of the MAX_EXEC_TIME we will stop processing the archive in this page
+	 * load, return the result to the client and wait for it to call us again so we can resume the extraction.
+	 *
+	 * This becomes important when the MAX_EXEC_TIME is close the the PHP, PHP-FPM or Apache timeout oon the server
+	 * (whichever is lowest) and there are fairly large files in the backup archive. If we start extracting a large,
+	 * compressed file close to a hard server timeout it's possible that we will overshoot that hard timeout and see the
+	 * extraction failing.
+	 *
+	 * Since Joomla Update is used to extract a ZIP archive with many small files we can keep at a fairly high 90%
+	 * without much fear that something will break.
+	 *
+	 * Example: if MAX_EXEC_TIME is 10 seconds and RUNTIME_BIAS is 80 each page load will take between 80% and 100% of
+	 * the MAX_EXEC_TIME, i.e. anywhere between 8 and 10 seconds.
+	 *
+	 * Lower values make it less overshooting MAX_EXEC_TIME when extracting large files.
+	 *
+	 * @var int
+	 */
+	private const RUNTIME_BIAS = 90;
+
+	/**
+	 * Minimum execution time (seconds).
+	 *
+	 * A request cannot take less than this many seconds. If it does, we add “dead time” (sleep) where the script does
+	 * nothing except wait. This is essentially a rate limiting feature to avoid hitting a server-side DoS protection
+	 * which could be triggered if we ended up sending too many requests in a limited amount of time.
+	 *
+	 * This should normally be less than MAX_EXEC * (RUNTIME_BIAS / 100). Values between that and MAX_EXEC_TIME have the
+	 * effect of almost always adding dead time in each request, unless a really large file is being extracted from the
+	 * ZIP archive. Values larger than MAX_EXEC will always add dead time to the request. This can be useful to
+	 * artificially reduce the CPU usage limit. Some servers might kill the request if they see a sustained CPU usage
+	 * spike over a short period of time.
+	 *
+	 * The chosen value of 3 seconds belongs to the first category, essentially making sure that we have a decent rate
+	 * limiting without annoying the user too much but also without catering for the most badly configured of shared
```

Comment:
```suggestion
	 * limiting without annoying the user too much but also catering for the most badly configured of shared
```

Did you mean to say without?